src/errors.rs

```diff
@@ -100,3 +100,59 @@ custom_error! {
 }
 
 impl WebSocketEvent for GatewayError {}
+
+custom_error! {
+    /// Voice Gateway errors
+    ///
+    /// Similar to [GatewayError].
+    ///
+    /// See <https://discord.com/developers/docs/topics/opcodes-and-status-codes#voice-voice-close-event-codes>;
+    #[derive(Clone, Default, PartialEq, Eq)]
+    pub VoiceGatewayError
+    // Errors we receive
+    #[default]
+    UnknownOpcode = "You sent an invalid opcode",
+    FailedToDecodePayload = "You sent an invalid payload in your identifying to the (Voice) Gateway",
+    NotAuthenticated = "You sent a payload before identifying with the (Voice) Gateway",
+    AuthenticationFailed = "The token you sent in your identify payload is incorrect",
+    AlreadyAuthenticated = "You sent more than one identify payload",
+    SessionNoLongerValid = "Your session is no longer valid",
+    SessionTimeout = "Your session has timed out",
+    ServerNotFound = "We can't find the server you're trying to connect to",
+    UnknownProtocol = "We didn't recognize the protocol you sent",
+    Disconnected = "Channel was deleted, you were kicked, voice server changed, or the main gateway session was dropped. Should not reconnect.",
+    VoiceServerCrashed = "The server crashed, try resuming",
+    UnknownEncryptionMode = "Server failed to decrypt data",
+
+    // Errors when initiating a gateway connection
+    CannotConnect{error: String} = "Cannot connect due to a tungstenite error: {error}",
+    NonHelloOnInitiate{opcode: u8} = "Received non hello on initial gateway connection ({opcode}), something is definitely wrong",
+
+    // Other misc errors
+    UnexpectedOpcodeReceived{opcode: u8} = "Received an opcode we weren't expecting to receive: {opcode}",
+}
+
+impl WebSocketEvent for VoiceGatewayError {}
+
+custom_error! {
+    /// Voice UDP errors.
+    #[derive(Clone, PartialEq, Eq)]
+    pub VoiceUdpError
+
+    // General errors
+    BrokenSocket{error: String} = "Could not write / read from udp socket: {error}",
```

Comment:
<!--
 Copyright (c) 2024 star
 
 This Source Code Form is subject to the terms of the Mozilla Public
 License, v. 2.0. If a copy of the MPL was not distributed with this
 file, You can obtain one at https://mozilla.org/MPL/2.0/.
-->


```suggestion
    BrokenSocket{error: String} = "Could not write / read from UDP socket: {error}",
```